Building-with-IntelliJ-IDEA.md

```diff
@@ -20,104 +21,143 @@ You can build RuneLite locally using [JDK 11](https://adoptium.net/temurin/relea
+Alternatively, you can select your run configuration in the dropdown box at the top and hit the ``Run`` button.
 
-![project-maven-refresh](https://i.imgur.com/L2SmWtF.png)
+![custom_configuration_top](img/building-with-intellij/custom_configuration_top.png)
+
+Other useful run configurations include:  
+``clean install -DskipTests -U`` which is your first troubleshooting step that you should run multiple times in case of a build failures  
+``-Dcheckstyle.skip=false checkstyle:check`` to only run checkstyle without building the client  
+``surefire:test`` to only run the tests without building the client  
+
+We now need to tell IntelliJ to pick up changes based on Maven build (it should do that automatically, but sometimes it doesn't). Click on `Generate Sources and Update folders for all projects` in the Maven window.
+
+![generate_sources.png](img/building-with-intellij/generate_sources.png)
 
 ## Running the project
 
-Now locate *Project* in sidebar and click that open project view, and expand the tree to `runelite/runelite-client/src/main/java/net/runelite/client`. And all you need to do now is right-click `RuneLite` class and select run, and you are done, RuneLite should open:
+Now locate `Project` in the sidebar, and in the project view expand the tree to `runelite/runelite-client/src/main/java/net/runelite/client`. 
+And all you need to do now is right-click the `RuneLite` class and select `Run 'RuneLite.main()'`
+
+![run_runelite.png](img/building-with-intellij/run_runelite.png)
+
+If you receive a notification regarding Lombok's annotation processing, press ``Enable annotation processing``:
+
+![lombok_annotation_processing](img/building-with-intellij/lombok_annotation_processing.png)
+
+In case of errors regarding missing classes, click the ``Reload All Maven Projects`` in the Maven tab.
+
+![reload_maven_projects](img/building-with-intellij/reload_maven_projects.png)
 
-![project-view-run-main](https://i.imgur.com/RXkMc48.png)
+And you are done! RuneLite should open.
 
-If you are having any issues with this step see [Troubleshooting](#troubleshooting) section.
+If you are having any issues with this step see the [Troubleshooting](#troubleshooting) section.
 
 ## Conclusion
 
-Success! You can switch between running the client and building the Maven project by switching between the **RuneLite** configuration and your custom Maven configuration (should be named *runelite-parent [install]* by default):
+Success! You can switch between running the client and building the Maven project by switching between the ``RuneLite`` configuration and your custom Maven configuration (e.g. ``runelite-parent [install]``):
 
-![select-run-config](https://i.imgur.com/zKSqojU.png)
+![select_run_config](img/building-with-intellij/select_run_config.png)
 
-# Troubleshooting
 
-## Missing git
+### WARNING: An illegal reflective access operation has occurred <!-- omit in toc -->
```

Comment:
This is in somewhat of a weird spot. Visually it just reads strangely for the section above to be congratulatory, but this to show up sort of out of nowhere. I'd expect this to be a `<h2>` (not omitted from toc) under the troubleshooting section.